expertise/create_dataset.py

```diff
@@ -46,22 +46,34 @@ def get_profile_ids(openreview_client, group_ids):
     :param group_ids: List of group ids
     :type group_ids: list[str]
 
-    :return: List of tilde ids
+    :return: List of tuples containing (tilde_id, email)
     :rtype: list
     """
-    tilde_members = []
+    members = []
     for group_id in group_ids:
         group = openreview_client.get_group(group_id)
+
         profile_members = [member for member in group.members if '~' in member]
+        profile_search_results = openreview_client.search_profiles(emails=None, ids=profile_members, term=None)
+        tilde_members = []
+        for profile in profile_search_results:
+            preferredEmail = profile.content.get('preferredEmail')
+            # If user does not have preferred email, use first email in the emailsConfirmed list
+            preferredEmail = preferredEmail or profile.content.get('emailsConfirmed') and len(profile.content.get('emailsConfirmed')) and profile.content.get('emailsConfirmed')[0]
+            # If the user does not have emails confirmed, use the first email in the emails list
+            preferredEmail = preferredEmail or profile.content.get('emails') and len(profile.content.get('emails')) and profile.content.get('emails')[0]
+            # If the user Profile does not have an email, use its Profile ID
+            tilde_members.append((profile.id, preferredEmail or profile.id))
+        members.extend(tilde_members)
+
         email_members = [member for member in group.members if '@' in member]
         profile_search_results = openreview_client.search_profiles(emails=email_members, ids=None, term=None)
+        email_profiles = []
+        for email, profile in profile_search_results.items():
```

Comment:
@carlosmondra I found another issue here caused by of the `client.search_profiles` method. In the case when there are no emails in the match group members, we would be passing an empty list in the emails arg to search profiles and in that the `profile_search_results` would be returned as an empty list and the `.items()` call on it would fail.
Maybe we can code the line 70 as `profile_search_results = openreview_client.search_profiles(emails=email_members, ids=None, term=None) if email_members else {}`
What do you think?

Reply:
Thanks Mohit! I think I didn't do that because `search_profiles` already returns `{}` when no Profiles are found. I tested this out, just in case.
```python
>>> c.search_profiles(emails=['mondragon52369@hotmail.com'])
{}
>>> c.search_profiles(emails=['carlos@openreview.net'])
{'carlos@openreview.net': Profile(id = '~Carlos_Mondragon1',number = None,tcdate = 1559740458953,tmdate = 1580485922603,referent = None,packaging = None,invitation = '~/-/profiles',readers = ['OpenReview.net', '~Carlos_Mondragon1'],nonreaders = None,signatures = ['~Carlos_Mondragon1'],writers = ['OpenReview.net'],content = {'preferredEmail': 'carlos@openreview.net', 'gender': 'Male', 'homepage': 'https://www.linkedin.com/in/carlosmondragonchapa/', 'emails': ['carlos@openreview.net'], 'names': [{'first': 'Carlos', 'middle': '', 'last': 'Mondragon', 'preferred': True, 'username': '~Carlos_Mondragon1'}], 'history': [{'position': 'MS student', 'start': 2017, 'end': 2019, 'institution': {'domain': 'umass.edu', 'name': 'University of Massachusetts, Amherst'}}], 'relations': [], 'emailsConfirmed': ['carlos@openreview.net']},metaContent = {'preferredEmail': {'signatures': ['~Carlos_Mondragon1']}, 'gender': {'signatures': ['~Carlos_Mondragon1']}, 'homepage': {'signatures': ['~Carlos_Mondragon1']}, 'emails': [{'signatures': ['~Carlos_Mondragon1']}], 'names': [{'signatures': ['~Carlos_Mondragon1']}], 'history': [{'signatures': ['~Carlos_Mondragon1']}]},active = True,password = True,tauthor = 'OpenReview.net')}
```

Reply:
Sorry if I wasn't clear earlier. Try calling `c.search_profiles(emails=[])` and that would return an empty list which would then cause an exception as the code would look for `.items()` on a list. This is what will happen at line 70 if `email_members` is an empty list.

Reply:
Done! I pushed the changes